RecoEgamma/PhotonIdentification/plugins/PhotonIDValueMapProducer.cc

```diff
@@ -1,686 +1,488 @@
+        vars[12].push_back(photonIsoSum);
+
+        // Worst isolation computed with no vetos or ptMin cut, as in Run 1 Hgg code.
+        unsigned char options = 0;
+        vars[13].push_back(computeWorstPFChargedIsolation(iPho, pfCandsHandle, vertices, pv, options));
+
+        // Worst isolation computed with cone vetos and a ptMin cut, as in Run 2 Hgg code.
+        options |= PT_MIN_THRESH | DR_VETO;
+        vars[14].push_back(computeWorstPFChargedIsolation(iPho, pfCandsHandle, vertices, pv, options));
+
+        // Like before, but adding primary vertex constraint
+        options |= PV_CONSTRAINT;
+        vars[15].push_back(computeWorstPFChargedIsolation(iPho, pfCandsHandle, vertices, pv, options));
+
+        // PFCluster Isolations
+        vars[16].push_back(iPho->trkSumPtSolidConeDR04());
+        if (isAOD_) {
+            vars[17].push_back(0.f);
+            vars[18].push_back(0.f);
+        } else {
+            edm::Ptr<pat::Photon> patPhotonPtr{ src->ptrAt(i) };
+            vars[17].push_back(patPhotonPtr->hcalPFClusterIso());
+            vars[18].push_back(patPhotonPtr->ecalPFClusterIso());
+        }
+
+    }
+
+    // write the value maps
+    for (int i = 0; i < nVars_; ++i) {
+        auto valMap = std::make_unique<edm::ValueMap<float>>();
+        edm::ValueMap<float>::Filler filler(*valMap);
+        filler.insert(src, vars[i].begin(), vars[i].end());
+        filler.fill();
+        iEvent.put(std::move(valMap), names[i]);
+    }
 }
 
-void PhotonIDValueMapProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
-  //The following says we do not know what parameters are allowed so do no validation
-  // Please change this to state exactly what you do use, even if it is no parameters
-  edm::ParameterSetDescription desc;
-  desc.setUnknown();
-  descriptions.addDefault(desc);
+void PhotonIDValueMapProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions)
+{
+    // The following says we do not know what parameters are allowed so do no validation
+    // Please change this to state exactly what you do use, even if it is no parameters
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("ebReducedRecHitCollection");
+    desc.add<edm::InputTag>("ebReducedRecHitCollectionMiniAOD");
+    desc.add<edm::InputTag>("eeReducedRecHitCollection");
+    desc.add<edm::InputTag>("eeReducedRecHitCollectionMiniAOD");
+    desc.add<edm::InputTag>("esReducedRecHitCollection", edm::InputTag("reducedEcalRecHitsES"));
+    desc.add<edm::InputTag>("esReducedRecHitCollectionMiniAOD", edm::InputTag("reducedEgamma:reducedESRecHits"));
+    desc.add<edm::InputTag>("vertices");
+    desc.add<edm::InputTag>("verticesMiniAOD");
+    desc.add<edm::InputTag>("src");
+    desc.add<edm::InputTag>("srcMiniAOD");
+    desc.add<edm::InputTag>("particleBasedIsolation");
+    desc.add<edm::InputTag>("pfCandidates");
+    desc.add<edm::InputTag>("pfCandidatesMiniAOD");
+    descriptions.addDefault(desc);
```

Comment:
is it practical to write a valid default config file here?
From git grep, I see only one instance of photonIDValueMapProducer.
It looks like PhotonIDValueMapProducer_cfi can be removed and replaced here by an auto-generated file based on the fillDescriptions

Reply:
What about these phase2 specific inputs? Wouldn't we have to keep at least part of the config file for that? https://github.com/cms-sw/cmssw/blob/master/RecoEgamma/PhotonIdentification/python/PhotonIDValueMapProducer_cfi.py

Sorry I'm not so familiar with the python configurations...

Reply:
era modifiers can be added in a cff file which will include an auto-generated photonIDValueMapProducer_cfi

Reply:
So when I search, on GitHub, I find a few occurrences of `PhotonIDValueMapProducer_cfi`. What should I do? Add the era modifier in every file where `PhotonIDValueMapProducer_cfi` would be imported? Thanks for your help!

Reply:
- ```descriptions.add("photonIDValueMapProducer")```
- mv PhotonIDValueMapProducer_cfi photonIDValueMapProducer_cff, start it with import of photonIDValueMapProducer_cfi
- replace all use cases of PhotonIDValueMapProducer_cfi by photonIDValueMapProducer_cff